packages/api/blocks-api/src/blocks/transformToSaveIndex/transformToSaveIndex.ts

```diff
@@ -1,15 +1,21 @@
 import { FlatBlocks } from "../../flat-blocks/flat-blocks";
-import { Block, BlockDataInterface, BlockIndex } from "../block";
+import { Block, BlockDataInterface, BlockIndex, BlockIndexItem } from "../block";
 
 export function transformToSaveIndex(block: Block, blockData: BlockDataInterface): BlockIndex {
     const flatBlocks = new FlatBlocks(blockData, { name: block.name, visible: true, rootPath: "root" }); // breadthFirst or depthFirst is equally ok for the app, but the tests for original version are written for depth first traversal
 
     return flatBlocks.depthFirst().map((c) => {
-        return {
+        const blockIndexItem: BlockIndexItem = {
             blockname: c.name,
             jsonPath: c.pathToString(),
             visible: c.visible,
-            ...c.block.indexData(),
         };
+
+        const indexData = c.block.indexData();
+        if (indexData.dependencies && indexData.dependencies?.length > 0) {
```

Comment:
This should check whether `indexData` is empty imo:
 `Object.keys(indexData).length > 0`

Reply:
currently we have dependencies by default:
```
indexData(): BlockIndexData {
        return { dependencies: [] };
    }
```

however that could be changed

Reply:
I removed the default dependencies. That doesn't make sense anymore. So this approach works